code/modules/projectiles/updated_projectiles/gun_helpers.dm

```diff
@@ -739,10 +739,25 @@ should be alright.
 		A.ui_action_click(usr, src)
 
 
+/obj/item/weapon/gun/verb/toggle_rail_attachment()
+	set category = "Weapons"
+	set name = "Toggle Rail Attachment"
+	set desc = "Uses the rail attachement currently attached to the gun."
+	set src = usr.contents
+
+	if(!usr)
+		return
+
+	var/obj/item/weapon/gun/G = get_active_firearm(usr)
+	if(!G)
+		return
+
+	var/obj/item/attachable/A = G.rail
```

Comment:
You're doing this proc from inside the gun type and you grab a potentialy different gun from the usr instead of having it use itself. Why?